### DISCUSSION TITLE
Pull request for Typescript Example?

### DISCUSSION THREAD

**Discussion by xiaooye** [nextauthjs/next-auth#977] [Ideas]
**Your question**
I saw the empty typescript example on there so I copied the original next-auth-example and added typescript on it. But then I checked the pull request and found someone have made a pull request and have not been approved for months. Do you want more features or a more complexed example? I would be glad to help.

**What are you trying to do**
Add Typescript for next-auth

**Comment by balazsorban44** [comment]
The TypeScript support for `next-auth` is a community effort right now, the types are at DefinitelyTyped. Although adding TS support to `next-auth` itself is a very welcome addition! Feel free to clone the existing PR, or create your own.

**Comment by ndom91** [comment]
@xiaooye the DefinitelyTyped `next-auth` package can be installed via:

```
npm install --save-dev @types/next-auth
```

And you can make a PR to them [here](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/next-auth).

Of course built-in types would make a great PR to this repository as well :)
